toolchain/lowering/lowering_function_context.h

```diff
@@ -68,6 +104,15 @@ class LoweringFunctionContext {
 
   llvm::IRBuilder<> builder_;
 
+  // A worklist for blocks we need to lower.
+  LoweringBlockWorklist block_worklist_;
+
+  // Maps a function's SemanticsIR blocks to lowered blocks.
+  llvm::DenseMap<SemanticsNodeBlockId, llvm::BasicBlock*> blocks_;
+
+  // The synthetic block we most recently created.
+  std::optional<llvm::BasicBlock*> synthetic_block_ = std::nullopt;
```

Comment:
Is there any overhead when using optional? Should this be `nullptr`?

Reply:
Switched to just a pointer.